### PR TITLE
Pa02

### PR DESCRIPTION
**Pull request by Havardei96**
<3

### CODE REVIEW

**Review by OlavEspenes — APPROVED** [review]
All good.

File: src/pa02/snakes_simulations.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+__email__ = 'havardei@nmbu.no, olaves@nmbu.no'
+
+import random
+
+
+class Board:
+    """
+    Defining the board with its snakes, ladders and the goal.
+    """
+    def __init__(self, *args, **kwargs):
```
Comment (huynhngoc) [inline src/pa02/snakes_simulations.py:13]: This is cheating. And you cannot customize your ladders / snakes / goal with this implementation.

File: src/pa02/snakes_simulations.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        self.lazy_move = False
+        self.goal = 90
+
+    def goal_reached(self, position):
+        """
+        Returns true if a player has reached or moved beyond the goal.
+        """
+        if position >= self.goal:
+            return True
+        return False
```
Comment (huynhngoc) [inline src/pa02/snakes_simulations.py:27]: Please have the habit of returning comparison when possible.
`return position >=self.goal`

File: src/pa02/snakes_simulations.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        self.extra_steps = extra_steps
+        self.went_down_in_last_move = False
+
+    def move(self):
+        extra = self.extra_steps if self.went_down_in_last_move else 0
+        self.position += random.randint(1, 6) + extra
+        adjustment = self._board.position_adjustment(self.position)
+        self.position += adjustment
+        self.went_down_in_last_move = adjustment < 0
+        self.num_moves += 1
```
Comment (huynhngoc) [inline src/pa02/snakes_simulations.py:77]: 👍  nice and clean.

File: src/pa02/snakes_simulations.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+            return True
+        return False
+
+    def position_adjustment(self, position):
+        """
+        Returns the number of positions the player must move forward or
+        backward in case of a ladder or a snake.
+        """
+        if position in self.snakes.keys():
+            self.resilient_move = True
```
Comment (huynhngoc) [inline src/pa02/snakes_simulations.py:35]: Not sure about your purpose of creating these attributes, but it doesn't really affect your implementation.

**Review by huynhngoc — APPROVED** [review]
Nice implementation, but need more tests for test for correctness in your code. Remember you have to ways more tests than this in the Jan/Jun project to pass the "final exam".

Anyway, approved.

**Pull request merged**